optimize.c

```diff
@@ -2183,6 +2189,7 @@ opt_init(opt_state_t *opt_state, struct icode *ic)
 	opt_state->edges = (struct edge **)calloc(opt_state->n_edges, sizeof(*opt_state->edges));
 	if (opt_state->edges == NULL) {
 		free(opt_state->blocks);
+		opt_state->blocks = NULL;
```

Comment:
Would it be cleaner to just remove the `free()` calls from before `opt_error()` calls and let `opt_cleanup()` do the work?

Reply:
Indeed, removing lines instead of adding them looks cleaner